markdown-cheat-sheet.md

```diff
@@ -1,121 +1,252 @@
+Text with no blank lines between is "joined together".
+To specify a paragraph, leave a blank line between the end of the first paragraph and the beginning of the next.
 
-## H2
+## Headings
 
-### H3
+```text
+# Sample H1
+
+## Sample H2
+
+### Sample H3
+```
+
+will produce
+
+# Sample H1
+
+## Sample H2
+
+### Sample H3
+
+Headings introduce paragraph breaks automatically. However, you should still put a blank line before and a blank line after a heading.
+
+Note that there should only be one `H1` and it should be the first line in the markdown file. The `H1` style is typically used as a document title.
+
+Heading styles which need to move "upwards" to a higher number, should move one value at a time (`H1` to `H2`, for example, but not `H1` to `H3`). There are no constraints moving "downwards". `H6` is the biggest number (smallest sized heading).
+
+---
+
+## Horizontal Rules
+
+A horizontal rule is created using three dashes (`---`) on a line by itself and surrounded by blank lines.
+
+---
+
+## Coding - Block
+
+<pre>
+```bash
+declare ENV=${4:-sandbox}
+
+case "$1 $2 $3" in
+
+  'build manifests for')
+    if [[ ! $(git status | tee /dev/tty | grep 'nothing to commit, working tree clean') ]]; then
+```
+</pre>
+
+will produce
+
+```bash
+declare ENV=${4:-sandbox}
+
+case "$1 $2 $3" in
+
+  'build manifests for')
+    if [[ ! $(git status | tee /dev/tty | grep 'nothing to commit, working tree clean') ]]; then
+```
+
+Note: You can specify alternative syntax highlighting based on the coding language, eg. `ruby`, `bash`, `php`, etc. by marking the *first set* of backticks as shown. If do not want syntax highlighting, use the \`\`\`text marker.
+
+Note: You should leave blank lines before the first and after the final markers.
+
+## Coding - In-line
+
+You can produce inline-code by wrapping the code fragment in single backticks:
+
+```text
+This is some code: `echo something`
+```
+
+will produce
 
-### Bold
+This is some code: `echo something`
 
-Here is some **bold text**
+---
 
-### Italic
+## Text Formatting
 
-Here is some *italicized text*
+**Bold Text** is produced using `**Bold Text**`
 
-### Blockquote
+*Italic Text* is produced using `*Italic Text*`
 
-> blockquote
+You may of course have ***bold and italic*** text.
 
-### Ordered List
+Text may be ~~struck through~~ using `~~this will be struck through~~`.
 
-1. First item
-2. Second item
-3. Third item
+---
 
-### Unordered List
+## Hyperlinks
 
-- First item
-- Second item
-- Third item
+GFMD will automatically detect URLs and convert them to links like this https://www.softwareone.com/. However, this also is a rule-breaking form. It's preferable to use the following:
 
-### Code
+```text
+This form is preferred: [Corporate Website](https://www.softwareone.com/)
+```
 
-`code`
+This form is preferred: [Corporate Website](https://www.softwareone.com/)
 
-### Horizontal Rule
+---
+
+## Escape Sequence
+
+You can escape characters which would otherwise be actioned using \\ eg. \\\`
```

Comment:
Perhaps - 'You can escape character which would otherwise be actioned using \\ (a backslash)'